packages/api/internal/dns/server.go

```diff
@@ -63,35 +81,127 @@ func (d *DNS) handleDNSRequest(w resolver.ResponseWriter, r *resolver.Msg) {
 				},
 			}
 
+			var ip net.IP
+
 			sandboxID := strings.Split(q.Name, "-")[0]
-			ip, found := d.get(sandboxID)
-			if found {
-				a.A = net.ParseIP(ip).To4()
-			} else {
-				a.A = net.ParseIP(defaultRoutingIP).To4()
+
+			if addr, found := d.getLocal(sandboxID); found {
+				// we have it cached locally, this is
+				// still fine.
+				a.A = net.ParseIP(addr).To4()
+			} else if d.redis != nil {
+				// TODO: at least log the error
+				res, err := d.redis.Get(ctx, d.getCacheKey(sandboxID)).Result()
+				if err == nil {
+					// TODO: do we need to do
+					// anything with the error or
+					// distinguish between "can't
+					// find" and "server error"
+
+					ip = net.ParseIP(res)
+
+					// TODO: should we add this to
```

Comment:
Can you test how fast is the Redis request, if there's like 1000 keys? If we don't store in cache, it's probably easier to reason about, but is it worth the extra time? Every request to sandbox will do DNS request.